src/scripting/game_lua_kernel.cpp

```diff
@@ -4024,6 +4025,36 @@ const game_events::queued_event & game_lua_kernel::get_event_info() {
 	return *queued_events_.top();
 }
 
+/**
+	* @brief takes a formated string and put the variables
+	* @example wesnoth.format("I have $gold, and I need $goldNeed", {['gold'] = '100', ['goldNeed'] = '50'})
```

Comment:
Actually it's `$|` (ie, empty variable name) for a literal '$', and the formula string substitution handles this.